src/spi/graph/GraphClient.ts

```diff
@@ -63,28 +60,63 @@ export interface GraphClient {
      * @param {Q} variables the variables to be used
      * @param {any} options to be passed onto the underlying implementation
      * @returns {Promise<T>}
-     *
-     * DEPRECATED: use mutate() instead
+     * @deprecated use mutate() instead
      */
     executeMutation<T, Q>(mutation: string, variables?: Q, options?: any): Promise<any>;
 }
 
+/**
+ * Options to be passed into GraphClient.query.
+ */
 export interface QueryOptions<Q> {
+
+    /** Optional GraphQL query string */
     query?: string;
+    /**
+     * Optional absolute or relative path to the '.graphql' query file.
+     * Relative paths will get resolved against the path of the calling script.
+     */
     path?: string;
+    /**
+     * Optional name of the query operation in a '.graphql' file located in 'graphql/query'.
```

Comment:
One must specify either `query` or `name`, correct?

Reply:
Add a interface-level comment to reflect that.